addons/point_of_sale/static/src/customer_display/customer_display.xml

```diff
@@ -24,29 +24,28 @@
                 <div t-else="" class="d-flex flex-column align-items-center justify-content-center h-100 w-100">
                     <h1 class="display-2 w-75 mt-2 fw-bold text-center text-muted">Thank you.</h1>
                 </div>
-                <div t-if="order.isScaleScreenVisible">
+                <div t-if="order.scaleData">
                     <t t-set-slot="header">
-                        <h4 class="modal-title text-break text-center w-100" t-att-class="{ 'me-auto': this.env.isSmall }">
-                            <t t-esc="'Scaling Product: ' + order.scaleData.productName || 'Unnamed Product'"/>
+                        <h4 class="modal-title text-break text-center w-100" t-att-class="{ 'me-auto': env.isSmall }">
+                            <t t-esc="'Weighing Product: ' + order.scaleData.name"/>
```

Comment:
```suggestion
                            <span class="me-2">Weighing Product:</span>
                            <t t-esc="order.scaleData.name"/>
```

can you try this? The idea is to make sure `Weighing Product:` is translated. Terms inside the template are always translated.